ethers-signers/src/trezor/app.rs

```diff
@@ -56,9 +61,41 @@ impl TrezorEthereum {
         Ok(())
     }
 
+    fn get_cached_session() -> Result<Option<Vec<u8>>, TrezorError> {
+        let mut session = [0; 32];
+
+        let path = env::current_dir()
+            .map_err(|e| TrezorError::CacheError(e.to_string()))?
+            .join("cache")
+            .join("trezor.session");
+
+        if let Ok(file) = fs::File::open(path) {
```

Comment:
should we return a `None` here if the file does not exist?
and an error if reading the file failed?

Reply:
done

although in the beginning, i was thinking of issuing a warning, without halting the execution